protocol/messenger_activity_center.go

```diff
@@ -318,11 +399,126 @@ func (m *Messenger) handleActivityCenterAccepted(state *ReceivedMessageState, me
 }
 
 func (m *Messenger) handleActivityCenterDismissed(state *ReceivedMessageState, message protobuf.SyncActivityCenterDismissed) error {
-	resp, err := m.DismissActivityCenterNotifications(context.TODO(), toHexBytes(message.Ids), false)
+	resp, err := m.DismissActivityCenterNotifications(context.TODO(), toHexBytes(message.Ids), message.Clock, false)
 
 	if err != nil {
 		return err
 	}
 
 	return state.Response.Merge(resp)
 }
+
+func (m *Messenger) handleSyncActivityCenterNotificationState(state *ReceivedMessageState, a *protobuf.SyncActivityCenterNotificationState) error {
+	s := &ActivityCenterState{
+		HasSeen:   a.HasSeen,
+		UpdatedAt: a.UpdatedAt,
+	}
+	_, err := m.persistence.UpdateActivityCenterNotificationState(s)
+	if err != nil {
+		return err
+	}
+	state.Response.SetActivityCenterState(s)
+	return nil
+}
+
+func (m *Messenger) handleSyncActivityCenterNotifications(state *ReceivedMessageState, a *protobuf.SyncActivityCenterNotifications) error {
+	var notifications []*ActivityCenterNotification
+	for _, n := range a.ActivityCenterNotifications {
+		notification, err := convertActivityCenterNotificationFromProtobuf(n)
+		if err != nil {
+			return err
+		}
+		err = m.persistence.SaveActivityCenterNotification(notification, false)
+		if err != nil {
+			return err
+		}
+		notifications = append(notifications, notification)
+	}
+	response, err := m.processActivityCenterNotifications(notifications, true)
```

Comment:
@qfrank the same issue is here, we're adding all notifications to the response, regardless they are updated in the db or skipped because of the older clock.